packages/utils/src/cache/local-cache.ts

```diff
@@ -0,0 +1,74 @@
+import type { CacheItem } from './interfaces/CacheItem';
+import type { CacheStorage } from './interfaces/CacheStorage';
+import { CacheIndexedDBStorage } from './cache-indexeddb.js';
+import { CacheLocalStorage } from './cache-localstorage.js';
+
+export type LocalCacheConfig = {
+  storage: ('localstorage' | 'indexeddb');
+};
+
+/**
+ * Stores data in a local cache that can be specified to be stored in localstorage or indexedDB.
+ */
+export class LocalCache {
+  /**
+   * Storage to store data
+   */
+  protected storage!: CacheStorage;
+
+  constructor (name: string, config: LocalCacheConfig) {
+    if (config.storage.toLowerCase() === 'indexeddb') {
+      this.storage = new CacheIndexedDBStorage(name);
+    }
+    if (config.storage.toLowerCase() === 'localstorage') {
+      this.storage = new CacheLocalStorage(name);
+    }
+  }
```

Comment:
As this is a public API, perform checks on params. Also options should be optional.
```suggestion
  constructor (name: string, config?: LocalCacheConfig) {
    if (typeof name !== 'string') {
      throw new TypeError('Expected name to be of type string');
    }
    if (name.length === 0) {
      throw new RangeError('Expected name to have a length');
    }
    const options = Object.assign({}, config);
    switch (options.storage) {
      case 'indexeddb':
        this.storage = new CacheIndexedDBStorage(name);
        break;
      case 'localstorage':
      default:
        this.storage = new CacheLocalStorage(name);
    }
    throw new RangeError('Unknown storage type');
```

Reply:
Updated.
I move `throw new TypeError('Unknown storage type');` to be default case from your suggest ` throw new RangeError('Unknown storage type');`. It will throws error when user input wrong storage. And also supports case `undefined` when user don't set the local cache option.